lib/SecretsCrypto/sf-secrets-manager.c

```diff
@@ -204,17 +308,25 @@ static gboolean _sf_secrets_manager_cancel_interaction(SfSecretsManager *manager
 {
     SfSecretsManagerPrivate *priv = sf_secrets_manager_get_instance_private(manager);
     SfSecretsInteractionRequest *request = g_hash_table_lookup(priv->requests, request_id);
+    SfSecretsResultCode result = SF_SECRETS_RESULT_CODE_SUCCEEDED;
+    SfSecretsError error = SF_SECRETS_ERROR_NO;
+    const gchar *message = "";
 
     if (!request)
         return FALSE;
 
-    g_hash_table_remove(priv->requests, request_id);
-    g_object_weak_unref(G_OBJECT(request), (GWeakNotify)_sf_secrets_manager_remove_request, manager);
-    g_object_set(request,
-            "invocation", invocation,
-            NULL);
+    g_object_add_weak_pointer(G_OBJECT(request), (gpointer *)&request);
```

Comment:
'request' is in stack. Is it really ok to return and leave a weak ref to it behind?

Reply:
Definitely not, this is a boog in the non-happy case.